### PR TITLE
ENH: natural sorting on private/public study tables

### LINKED ISSUE #831: Samples column should not be sorted lexicographically
In the study results table (such as http://qiita.colorado.edu/study/public/), the Samples column reports the number of samples in the study. It currently sorts lexicographically, but this is not very useful. It should sort numerically.

### PR DESCRIPTION
**Pull request by adamrp**
Adds natural sorting plugin for dataTables, and applies natural sorting to private and public study tables.

Fixes #831

Replaces #862 (sorry, I couldn't git whisper a pull against @wasade's branch, so just created a new PR)

### CODE REVIEW

**Comment by wasade** [comment]
No worries! These connection resets are getting very annoying...

**Comment by adamrp** [comment]
Restarting build, I think it was just a stochastic Travis failure

**Comment by wasade** [comment]
Ya, already restarted it once

**Comment by adamrp** [comment]
This should be ready to go

File: qiita_pet/static/vendor/js/jquery.dataTables.plugin.natural.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ *       columnDefs: [
+ *         { type: 'natural', targets: 0 }
+ *       ]
+ *    } );
+ */
+
+(function() {
+
+/*
+ * Natural Sort algorithm for Javascript - Version 0.7 - Released under MIT license
```
Comment (ElDeveloper) [inline qiita_pet/static/vendor/js/jquery.dataTables.plugin.natural.js]: I think we may want to put a license file inside the `qiita_pet/static/vendor/licences/` folder.

File: qiita_pet/templates/private_studies.html [hunk 2]
```diff
@@ -3,13 +3,20 @@
...
+<script src="/static/vendor/js/jquery.dataTables.plugin.natural.js"></script>
 
 <script type="text/javascript">
 var current_study;
 
 $(document).ready(function() {
-    $('#user-studies-table').dataTable({"order": [[ 1, "asc" ]]});
-    $('#shared-studies-table').dataTable({"order": [[ 1, "asc" ]]});
+        $('#user-studies-table').dataTable({
+            order: [[ 1, "asc" ]],
```
Comment (ElDeveloper) [inline qiita_pet/templates/private_studies.html]: This 1 has to be changed to a 0. Such that the function sorts using the study title when it the table is first loaded? Same for the rest of the the occurrences of this index.

**Comment by adamrp** [comment]
Thanks @ElDeveloper, I added a couple more commits.

**Comment by adamrp** [comment]
Travis build keeps failing, but I'll keep restarting it...

**Comment by adamrp** [comment]
Okay, this should be ready!

**Comment by ElDeveloper** [comment]
Thanks @adamrp! I'm seeing something rather odd, not sure if you can reproduce this on your end, see this animated image:

issue_831

If you see, the order doesn't seem to be right at all, or am I missing something completely obvious here?

**Comment by antgonza** [comment]
Ping @adamrp, comments from @ElDeveloper

**Comment by adamrp** [comment]
Took me a while to figure this out... it's sorting that column based on the HTML, not the actual study title. the natsort plugin is not compatible with table entries that have HTML, so natsort is going to be limited to just columns 1,2,3 and 7.

**Comment by adamrp** [comment]
I *think* the current failure is due to the IPython 3.0 issue that's already been discussed.

**Comment by wasade** [comment]
First one is, @josenavas, able to look? Its on the workflow object, looks like `.pyout` may no longer exist. The second failure looks like something else tho?

**Comment by josenavas** [comment]
There is a PR that changes the ipython to use 2.4.1. My recommendation is to merge that PR first until we modify moi to support Ipython 3. I don't have time to take a look now.

**Comment by adamrp** [comment]
Restarted the build

**Comment by wasade** [comment]
There hasn't been any evidence yet that there is an issue with moi. @squirrelo's traceback looked like an environmental issue, and this appears to be an issue with the parallel workflow.

**Comment by adamrp** [comment]
Okay, as far as I can tell, this is now fixed, including the bug that @ElDeveloper recognized. Can someone verify?

**Comment by ElDeveloper** [comment]
:+1: I can verify this works as expected! Thanks @adamrp very useful!!

**Comment by adamrp** [comment]
Anyone want to merge?

**Pull request merged**